### PR TITLE
[onert] Introduce fine tuning feature for training

### PR DESCRIPTION
**Pull request by mbencer**
This PR adds fine tuning feature to onert. The new methods to nnfw API were added. onert_train tool was also extended to support this feature via `--frozen_ops_idx` argument.

TEST ENVIRONMENT:
- [test models](https://github.com/mbencer/OneLearningNotes/tree/mbencer/FineTuning/test_models)
- [training data](https://github.com/mbencer/OneLearningNotes/tree/mbencer/FineTuning/test_models)
- [C++ tool to compare weights before and after training](https://github.com/mbencer/OneLearningNotes/tree/mbencer/FineTuning/circle_compare_weights)
- [Python script](https://github.com/mbencer/OneLearningNotes/blob/mbencer/FineTuning/report.py) which run training for different `epoch`/`batch size`/`frozen ops` config to measure time and accuracy

**The test methodology was described [here](https://github.com/mbencer/OneLearningNotes/blob/mbencer/FineTuning/FineTuning.md) and the results are collected in [tests report](https://github.com/mbencer/OneLearningNotes/blob/mbencer/FineTuning/report.md)**

Issue: https://github.com/Samsung/ONE/issues/12386

### CODE REVIEW

File: runtime/onert/api/nnfw/src/nnfw_api_internal.cc [hunk 1]
```diff
@@ -1695,7 +1742,7 @@ NNFW_STATUS nnfw_session::train_export_circle(const char *path)
 
       auto subg = subgs->Get(0); // Get 1st subgraph
       if (!idx.valid() || idx.value() >= subg->tensors()->size())
-        throw std::runtime_error("Trainable tensor index is out of range");
+        return;
```
Comment (mbencer) [inline runtime/onert/api/nnfw/src/nnfw_api_internal.cc:1745]: It's just a workaround for a problem reproduced even on master to enable exporting trained circle to a file (used in my test scripts to compare weights before and after training).
My assumption is that the trained graph can have more tensors because of the tensors consumed by a Loos function, but not sure how it worked before.

BTW:
```
 auto model = ::circle::GetModel(mmapfile.buf());
      if (!model)
        throw std::runtime_error("Failed to get model from circle");
      auto subgs = model->subgraphs();
      if (!subgs || subgs->size() != 1)
        throw std::runtime_error("Circle does not has valid subgraph or has multiple subgraphs");

      auto subg = subgs->Get(0); // Get 1st subgraph
```
can be move outside `iterateTrainableTensors` but I'll try to deal with it in a separate PR.

File: runtime/onert/api/nnfw/src/nnfw_api_internal.cc [hunk 2]
```diff
@@ -1545,6 +1545,53 @@ NNFW_STATUS nnfw_session::train_set_output(uint32_t index, NNFW_TYPE /*type*/, v
...
+
+  if (op_index >= primary_subgraph()->operations().size())
+  {
+    std::cerr << "Error during nnfw_session::nnfw_train_disable_node_update: provided op_index="
+              << op_index << " is out of operators range" << std::endl;
+    return NNFW_STATUS_INVALID_STATE;
+  }
+
+  const auto ir_op_index = onert::ir::OperationIndex{op_index};
+  auto &options = _coptions[0];
```
Comment (jyoungyun) [inline runtime/onert/api/nnfw/src/nnfw_api_internal.cc:1565]: How about using `TrainingInfo` instead of _coptions?

Comment (mbencer) [reply runtime/onert/api/nnfw/src/nnfw_api_internal.cc]: @jyoungyun I like this idea, introduced ;)

**Comment by mbencer** [comment]
The feature was implemented
